runtime/integration-tests/src/generic/cases/proxy.rs

```diff
@@ -0,0 +1,236 @@
+{
+	assert!(configure_proxy_and_x_transfer::<T>(altair_runtime::ProxyType::Invest).is_err());
+}
+
+fn centrifuge_transfer_with_proxy_transfer<T: Runtime>()
+where
+	T: pallet_proxy::Config<ProxyType = centrifuge_runtime::ProxyType>,
+{
+	assert!(configure_proxy_and_transfer::<T>(centrifuge_runtime::ProxyType::Transfer).is_ok());
+}
+
+fn centrifuge_transfer_with_proxy_borrow<T: Runtime>()
+where
+	T: pallet_proxy::Config<ProxyType = centrifuge_runtime::ProxyType>,
+{
+	assert!(configure_proxy_and_transfer::<T>(centrifuge_runtime::ProxyType::Borrow).is_err());
+}
+
+fn centrifuge_transfer_with_proxy_invest<T: Runtime>()
+where
+	T: pallet_proxy::Config<ProxyType = centrifuge_runtime::ProxyType>,
+{
+	assert!(configure_proxy_and_transfer::<T>(centrifuge_runtime::ProxyType::Invest).is_err());
+}
+
+fn centrifuge_x_transfer_with_proxy_transfer<T: Runtime>()
+where
+	T: pallet_proxy::Config<ProxyType = centrifuge_runtime::ProxyType>,
+{
+	assert!(configure_proxy_and_x_transfer::<T>(centrifuge_runtime::ProxyType::Transfer).is_ok());
+}
+
+fn centrifuge_x_transfer_with_proxy_borrow<T: Runtime>()
+where
+	T: pallet_proxy::Config<ProxyType = centrifuge_runtime::ProxyType>,
+{
+	assert!(configure_proxy_and_x_transfer::<T>(centrifuge_runtime::ProxyType::Borrow).is_err());
+}
+
+fn centrifuge_x_transfer_with_proxy_invest<T: Runtime>()
+where
+	T: pallet_proxy::Config<ProxyType = centrifuge_runtime::ProxyType>,
+{
+	assert!(configure_proxy_and_x_transfer::<T>(centrifuge_runtime::ProxyType::Invest).is_err());
+}
+
+crate::test_for_runtimes!([development], development_transfer_with_proxy_transfer);
+crate::test_for_runtimes!([development], development_transfer_with_proxy_borrow);
+crate::test_for_runtimes!([development], development_transfer_with_proxy_invest);
+crate::test_for_runtimes!([development], development_x_transfer_with_proxy_transfer);
+crate::test_for_runtimes!([development], development_x_transfer_with_proxy_borrow);
+crate::test_for_runtimes!([development], development_x_transfer_with_proxy_invest);
+
+crate::test_for_runtimes!([altair], altair_transfer_with_proxy_transfer);
+crate::test_for_runtimes!([altair], altair_transfer_with_proxy_borrow);
+crate::test_for_runtimes!([altair], altair_transfer_with_proxy_invest);
+crate::test_for_runtimes!([altair], altair_x_transfer_with_proxy_transfer);
+crate::test_for_runtimes!([altair], altair_x_transfer_with_proxy_borrow);
+crate::test_for_runtimes!([altair], altair_x_transfer_with_proxy_invest);
+
+crate::test_for_runtimes!([centrifuge], centrifuge_transfer_with_proxy_transfer);
+crate::test_for_runtimes!([centrifuge], centrifuge_transfer_with_proxy_borrow);
+crate::test_for_runtimes!([centrifuge], centrifuge_transfer_with_proxy_invest);
+crate::test_for_runtimes!([centrifuge], centrifuge_x_transfer_with_proxy_transfer);
+crate::test_for_runtimes!([centrifuge], centrifuge_x_transfer_with_proxy_borrow);
+crate::test_for_runtimes!([centrifuge], centrifuge_x_transfer_with_proxy_invest);
```

Comment:
Sorry for this boilerplate, but we have no choice because each `ProxyType` is defined in each runtime, so they're different types. They can not be moved to runtime-common because the implementation uses the `RuntimeCall` of each runtime.

Actually, I'm thinking of moving the [generic runtime config](https://github.com/centrifuge/centrifuge-chain/blob/main/runtime/integration-tests/src/generic/config.rs#L45) used in integration tests to `runtime-common` to allow building things like that without copy-paste in all runtime, WDYT? @mustermeiszer @cdamian

Reply:
Tracking issue: https://github.com/centrifuge/centrifuge-chain/issues/1664